projects/angular/src/data/datagrid/render/header-renderer.ts

```diff
@@ -63,20 +60,29 @@ export class DatagridHeaderRenderer implements OnDestroy {
     this.columnsService.columns[index] = this.columnState;
   }
 
-  private stateChanges(state: ColumnState) {
-    if (state.changes && state.changes.length) {
-      state.changes.forEach(change => {
-        switch (change) {
-          case DatagridColumnChanges.WIDTH:
-            this.setWidth(state);
-            break;
-          case DatagridColumnChanges.HIDDEN:
-            this.setHidden(state);
-            break;
-          default:
-            break;
-        }
-      });
+  setWidth(state: ColumnState) {
+    if (state.strictWidth) {
+      if (this.columnResizerService.resizedBy) {
+        this.resizeEmitter.emit(state.width);
+        this.renderer.setStyle(this.el.nativeElement, 'width', state.width + 'px');
+        this.widthSet = false;
+      }
+      // Don't set width if there is a user-defined one. Just add the strict width class.
+      this.renderer.addClass(this.el.nativeElement, STRICT_WIDTH_CLASS);
+      this.autoSet = false;
+    } else {
+      this.renderer.removeClass(this.el.nativeElement, STRICT_WIDTH_CLASS);
+      this.renderer.setStyle(this.el.nativeElement, 'width', state.width + 'px');
+      this.widthSet = true;
+      this.autoSet = true;
+    }
+  }
+
+  setHidden(state: ColumnState) {
+    if (state.hidden) {
+      this.renderer.addClass(this.el.nativeElement, HIDDEN_COLUMN_CLASS);
+    } else {
+      this.renderer.removeClass(this.el.nativeElement, HIDDEN_COLUMN_CLASS);
```

Comment:
Is there a reason behind the move of `setWidth` and `setHidden` from their original places?

Reply:
They are public now so we can access them from main-renderer and we have a specific order for class members otherwise the lint will blow up.